internal/sourcemap/chained.go

```diff
@@ -35,10 +36,15 @@ func (c ChainedFetcher) Fetch(ctx context.Context, name, version, path string) (
 	var lastErr error
 	for _, f := range c {
 		consumer, err := f.Fetch(ctx, name, version, path)
+		if !errors.Is(err, ErrFetcherUnvailable) {
+			return consumer, err
+		}
```

Comment:
Shouldn't this be 
```

if err != nil {
  if errors.Is(err, ErrFetcherUnvailable){
    return nil, err
  }
  lastErr = err
  continue
}

if consumer != nil{
  return consumer, nil
}



```

Reply:
The idea is to always return what we get when the fetcher worked.

worked= it returns either a sourcemap or an error except `ErrFetcherUnvailable`

If we get `ErrFetcherUnvailable` then the fetcher is not available (e.g. ES not reachable) and we try the next chained fetcher.
I've pushed a new change to simplify and document the logic a bit better.